### PR TITLE
Custom configs

### PR DESCRIPTION
**Pull request by Alex-Lewandowski**
- Add info on using custom envs and shell scripts
- Tolerate yamls with `.yml` or `.yaml` file extensions
- Add `minimal_notebook_env.yaml` example
- Move all shell scripts into `Scripts` directory
- Add .gitignore

### CODE REVIEW

**Review by dgpalmieri — APPROVED** [review]
A couple (very) minor comments, but this looks great!

File: Create_OSL_Conda_Environments.ipynb [hunk 1]
```diff
@@ -127,24 +165,28 @@
...
+    "# Creates the environment\n",
     "!mamba env create -f $yml --prefix $prefix -q --force\n",
     "\n",
+    "# Without running this, every environment would appear in the GUI with the same name, 'Python3'\n",
     "!mamba run -n $name kernda --display-name $name -o $prefix/share/jupyter/kernels/python3/kernel.json\n",
     "\n",
+    "# Runs any associated shell scripts\n",
     "if name == \"insar_analysis\":\n",
-    "    !source install_insar_analysis_pkgs.sh\n",
+    "    !source \"Scripts/install_insar_analysis_pkgs.sh\"\n",
```
[the remark this answers is not in the document]
Comment (Alex-Lewandowski) [reply Create_OSL_Conda_Environments.ipynb:176]: We could but I think I'd still want to leave the full relative path in there so the repo is transportable. We can't assume someone who has cloned the repo also put the `Scripts` dir on path.

File: Create_OSL_Conda_Environments.ipynb [hunk 2]
```diff
@@ -55,7 +88,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "env_ymls = glob.glob(\"/home/jovyan/conda_environments/Environment_Configs/*.yml\")\n",
+    "env_ymls = glob.glob(\"/home/jovyan/conda_environments/Environment_Configs/*.y*ml\")\n",
```
[the remark this answers is not in the document]
Comment (Alex-Lewandowski) [reply Create_OSL_Conda_Environments.ipynb:91]: Good catch! I'll switch it over to Pathlib. In writing the above comment about putting `Scripts` on path, I realized we also have the `Scripts` path hard-coded for the glob, which breaks portability. I'll use `Path.cwd().rglob()` to take care of that.

**Review by dgpalmieri — APPROVED** [review]
LGTM!

**Pull request merged**